src/coreclr/vm/object.cpp

```diff
@@ -1740,7 +1739,7 @@ BOOL Nullable::UnBoxNoGC(void* destPtr, OBJECTREF boxedVal, MethodTable* destMT)
         {
             // For safety's sake, also allow true nullables to be unboxed normally.
             // This should not happen normally, but we want to be robust
-            if (destMT == boxedVal->GetMethodTable())
+            if (destMT->IsEquivalentTo(boxedVal->GetMethodTable()))
```

Comment:
Was this fixing a pre-existing bug ?

Reply:
No it wasn't - I reverted the change. It came about because I was using that method at one point and it wasn't consistent with the similar `Nullable::UnBox()` that does use `IsEquivalentTo` so it seemed safer to change it. However, I reverted it since I'm not sure if it's needed plus could be a bit slower.